main.tf

```
@@ -122,6 +123,13 @@ resource "aws_lambda_function" "default" {
   timeout                        = var.timeout
   tags                           = var.tags
 
+  dynamic "dead_letter_config" {
+    for_each = local.dead_letter_config
+    content {
+      target_arn = var.dead_letter_target_arn
+    }
+  }
```

Please see the other dynamic block and try to follow the same style:
```suggestion
  dynamic "dead_letter_config" {
    for_each = local.dead_letter_config
    
    content {
      target_arn = var.dead_letter_target_arn
    }
  }
```

---

Updated :)